### PR TITLE
fix: rdf-js parsers actually return a quad stream

### PR DESCRIPTION
**Pull request by tpluscode**
The base `EventEmitter` is not enough to write the code below:

```typescript
import stringToStream from 'string-to-stream'
import rdf from 'rdf-ext'
import { Dataset } from 'rdf-js'

export function parse(quads: string): Promise<Dataset> {
  const stream = stringToStream(quads) as any

  return rdf.dataset().import(parser.import(stream))
}
```

This code works fine and for typings to be correct the `parser.stream()` should return a stream

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: _see above_
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/rdfjs__parser-jsonld/rdfjs__parser-jsonld-tests.ts [hunk 1]
```diff
@@ -14,6 +13,6 @@ const parser3 = new Parser({ baseIRI, context, factory });
 
 const sink: Sink = parser1;
 
-const eventEmitter1: EventEmitter = parser1.import(stream);
-const eventEmitter2: EventEmitter = parser1.import(stream, {});
-const eventEmitter3: EventEmitter = parser1.import(stream, { baseIRI, context, factory });
+const eventEmitter1: Stream = parser1.import(stream);
+const eventEmitter2: Stream = parser1.import(stream, {});
+const eventEmitter3: Stream = parser1.import(stream, { baseIRI, context, factory });
```
Comment (thewilkybarkid) [inline types/rdfjs__parser-jsonld/rdfjs__parser-jsonld-tests.ts]: Should also test with a different quad implementation.

Comment (tpluscode) [reply types/rdfjs__parser-jsonld/rdfjs__parser-jsonld-tests.ts]: Something like this?
Have to go all the way back to a custom quad. Using `BaseQuad` is cheating imo ;)

```ts
interface SpecializedQuad extends BaseQuad {}
const typedStream: Stream<SpecializedQuad> = <any> {};
const typedParser: Parser<SpecializedQuad> = <any> {};
const typedImported: Stream<SpecializedQuad> = typedParser.import(typedStream);
const typedImported1: Stream<SpecializedQuad> = typedParser.import(typedStream, {});
const typedImported2: Stream<SpecializedQuad> = typedParser.import(typedStream, { baseIRI, context, factory });
```

Comment (thewilkybarkid) [reply types/rdfjs__parser-jsonld/rdfjs__parser-jsonld-tests.ts]: Yep!

File: types/rdfjs__serializer-jsonld-ext/index.d.ts [hunk 2]
```diff
@@ -19,10 +19,10 @@ declare namespace Serializer {
     }
 }
 
-declare class Serializer implements Sink {
+declare class Serializer<Q extends BaseQuad = Quad> implements Sink<Q> {
     constructor(options?: Serializer.SerializerOptions);
 
-    import(stream: Stream, options?: Serializer.SerializerOptions): EventEmitter;
+    import(stream: Stream<Q>, options?: Serializer.SerializerOptions): EventEmitter;
```
Comment (thewilkybarkid) [inline types/rdfjs__serializer-jsonld-ext/index.d.ts]: All extensions of `@rdfjs/sink` return a `Stream`?

Comment (tpluscode) [reply types/rdfjs__serializer-jsonld-ext/index.d.ts]: I don't understand

Comment (thewilkybarkid) [reply types/rdfjs__serializer-jsonld-ext/index.d.ts]: Sorry, mixed terminologies. Meant a `Readable` stream. But have just looked at the code and see `read` is a dummy function so no need to change it here.

**Review by thewilkybarkid — APPROVED** [review]

**Pull request merged**